### PR TITLE
TASK-41619 : Synchronisation AD - User status

### PR DESCRIPTION
**Pull request by GouadriaHanen**
added enabled to attributes while retrieving users from AD

### CODE REVIEW

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/IDMUserListAccess.java [hunk 1]
```diff
@@ -100,11 +100,11 @@ public IDMUserListAccess(UserQueryBuilder userQueryBuilder, int pageSize, boolea
         if (fullResults == null) {
             users = listQuery(index, length);
 
-            if ((this.userStatus == UserStatus.ENABLED || this.userStatus == UserStatus.DISABLED) && !isDBOnly) {
+            if ((this.userStatus == UserStatus.ENABLED || this.userStatus == UserStatus.DISABLED)) {
```
Comment (ahamdi) [inline component/identity/src/main/java/org/exoplatform/services/organization/idm/IDMUserListAccess.java]: Why did you remove the boolean isDbOnly ?

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/IDMUserListAccess.java [hunk 2]
```diff
@@ -222,7 +222,7 @@ PicketLinkIDMOrganizationServiceImpl getOrganizationService() {
             while (iterator.hasNext() && result.size() < length) {
               org.picketlink.idm.api.User user = iterator.next();
                 Attribute attr = getIDMService().getIdentitySession().getAttributesManager().getAttribute(user.getKey(), EntityMapperUtils.USER_ENABLED);
-                if ((userStatus == UserStatus.ENABLED && attr == null) || (userStatus == UserStatus.DISABLED && attr != null && attr.getValue().toString().equals("false"))) {
+                if ((userStatus == UserStatus.ENABLED && attr == null) || (userStatus == UserStatus.DISABLED && attr != null && attr.getValue().toString().equals("false"))|| (userStatus == UserStatus.ENABLED && attr != null && attr.getValue().toString().equals("true"))) {
```
Comment (ahamdi) [inline component/identity/src/main/java/org/exoplatform/services/organization/idm/IDMUserListAccess.java]: Could you combine this check with the first one in the same instruction ?

File: component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java [hunk 3]
```diff
@@ -793,7 +793,7 @@ exclude USER type (search for all group types) */
                   //Nothing
                }
                /** Begin eXo customization : PLF-7270**/
-               if (entry.getValue() == null || entry.getValue().length == 0) {
+               if ((entry.getValue() == null || entry.getValue().length == 0) && mappedAttributeName!= "enabled") {
```
Comment (ahamdi) [inline component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java]: Could you tell more about this check ?

File: component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java [hunk 4]
```diff
@@ -3256,54 +3256,39 @@ private void filterByAttributesValues(Collection<IdentityObject> objects, Map<St
                //Nothing
             }
 
-            // If the attribute key is enable, consider its absence as it was equals to true
```
Comment (ahamdi) [inline component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java]: Could youi tell more about this change ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/IDMUserListAccess.java (hunk 1 above)
Comment (GouadriaHanen) [reply component/identity/src/main/java/org/exoplatform/services/organization/idm/IDMUserListAccess.java]: removed to always filter by status when retrieving by status

File: component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java (hunk 3 above)
Comment (GouadriaHanen) [reply component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java]: "enabled" is added to attributes so the query becomes " and not exists(from io.attributes as attrTableJoinName where attrTableJoinName.name = :enabled " will return an empty slice

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/IDMUserListAccess.java (hunk 2 above)
Comment (GouadriaHanen) [reply component/identity/src/main/java/org/exoplatform/services/organization/idm/IDMUserListAccess.java]: No, it is not the same

File: component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java (hunk 4 above)
Comment (GouadriaHanen) [reply component/identity/src/main/java/org/picketlink/idm/impl/store/hibernate/PatchedHibernateIdentityStoreImpl.java]: an internal user may have its status null means that he is enabled

**Review by ahamdi — APPROVED** [review]

**Pull request merged**